tests/drivers/hwinfo/api/src/main.c

```diff
@@ -84,13 +84,140 @@ static void test_device_id_enotsup(void)
 }
 #else
 static void test_device_id_enotsup(void) {}
-#endif /* CONFIG_HWINFO_HAS_DRIVER not defined*/
+#endif /* CONFIG_HWINFO_DEVICE_ID_HAS_DRIVER not defined*/
+
+/*
+ * @addtogroup t_hwinfo_get_reset_cause_api
+ * @{
+ * @defgroup t_hwinfo_get_reset_cause test_hwinfo_get_reset_cause
+ * @brief TestPurpose: verify get reset cause works.
+ * @details
+ * - Test Steps
+ *   -# Set target buffer to a known value
+ *   -# Read the reset cause
+ *   -# Check if target buffer has been altered
+ * - Expected Results
+ *   -# Target buffer contents should be changed after the call.
+ * @}
+ */
+#ifdef CONFIG_HWINFO_RESET_CAUSE_HAS_DRIVER
+static void test_get_reset_cause(void)
+{
+	u32_t cause;
+	ssize_t ret;
+
+	/* Set `cause` to a known value prior to call. */
+	cause = 0xDEADBEEF;
+
+	ret = hwinfo_get_reset_cause(&cause);
+	/* If a platform does not support reset cause, pass the test. */
+	if (ret == -ENOTSUP) {
```

Comment:
The driver should not return -ENOTSUP.
This is the test for `CONFIG_HWINFO_RESET_CAUSE_HAS_DRIVER`
Please make this an assert

Reply:
Right. This was before we had a `CONFIG_` for each feature. I'll make the change.